From 98d27d303621db542e8c023b5dbd145b94b86653 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Thu, 8 Nov 2018 15:12:25 +0800
Subject: [PATCH 5067/5242] MLK-20277-2: ASoC: rpmsg_ak4497: remove
 digital_mute function

commit  2d8a7e59901f2a7bb373af49d2ea4c1e02f452a6 from
https://source.codeaurora.org/external/imx/linux-imx.git

For the LPA mode, when the system enter suspend, the M4 will
continue to play the data, but for normal ALSA case, the digital
mute should be called at suspend, so the codec will be mute,
which conflict with the requirement of LPA.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/rpmsg_ak4497.c |   28 ----------------------------
 1 file changed, 28 deletions(-)

diff --git a/sound/soc/codecs/rpmsg_ak4497.c b/sound/soc/codecs/rpmsg_ak4497.c
index 78a78ad..affa06b 100644
--- a/sound/soc/codecs/rpmsg_ak4497.c
+++ b/sound/soc/codecs/rpmsg_ak4497.c
@@ -739,33 +739,6 @@ static int ak4497_set_bias_level(struct snd_soc_component *component,
 	return 0;
 }
 
-static int ak4497_set_dai_mute(struct snd_soc_dai *dai, int mute)
-{
-	struct snd_soc_codec *codec = dai->codec;
-	struct rpmsg_ak4497_priv *ak4497 = snd_soc_codec_get_drvdata(codec);
-	int nfs, ndt;
-
-	nfs = ak4497->fs1;
-
-	if (mute) { /* SMUTE: 1 , MUTE */
-		snd_soc_update_bits(codec, AK4497_01_CONTROL2, 0x01, 0x01);
-		ndt = 7424000 / nfs;
-		mdelay(ndt);
-
-		/* External Mute ON */
-		if (gpio_is_valid(ak4497->mute_gpio))
-			gpio_set_value_cansleep(ak4497->mute_gpio, 1);
-	} else { /* SMUTE: 0, NORMAL operation */
-
-		/* External Mute OFF */
-		if (gpio_is_valid(ak4497->mute_gpio))
-			gpio_set_value_cansleep(ak4497->mute_gpio, 0);
-		snd_soc_update_bits(codec, AK4497_01_CONTROL2, 0x01, 0x00);
-	}
-
-	return 0;
-}
-
 #define AK4497_RATES (SNDRV_PCM_RATE_8000 | SNDRV_PCM_RATE_11025 |\
 		      SNDRV_PCM_RATE_16000 | SNDRV_PCM_RATE_22050 |\
 		      SNDRV_PCM_RATE_32000 | SNDRV_PCM_RATE_44100 |\
@@ -807,7 +780,6 @@ static int ak4497_startup(struct snd_pcm_substream *substream,
 	.hw_params	= ak4497_hw_params,
 	.set_sysclk	= ak4497_set_dai_sysclk,
 	.set_fmt	= ak4497_set_dai_fmt,
-	.digital_mute   = ak4497_set_dai_mute,
 };
 
 struct snd_soc_dai_driver rpmsg_ak4497_dai[] = {
-- 
1.7.9.5

